### PR TITLE
add alienswap dex contract

### PR DESCRIPTION
**Pull request by xcaptain**
Our contract is at: https://etherscan.io/address/0x83746de31fc8de985ffe46c1c20ea6d7d8f4ed3a
please review.

### CODE REVIEW

**Comment by xcaptain** [comment]
@liutaurasjanuskas please take a look

File: src/adapters/alienswap-ethereum-1/index.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            nfts: [
+                {
+                    id: tokenId,
+                    amount: Number(amount),
+                    contract: nftContract.toLowerCase(),
+                },
+            ],
+            token: this.token,
+            tokenSymbol: symbol?.symbol || "",
+            price: nativePrice.toNumber(),
```
Comment (liutaurasjanuskas) [inline src/adapters/alienswap-ethereum-1/index.ts:80]: Price is incorrect. For e.g. if a single transaction holds multiple sales - the volume would be doubled.

Example transaction: https://etherscan.io/tx/0xa79a2d5ebf9a8b145ae2acdf6ab16084a2c60a0b1fcd9d5c930e9163784b4e90
It should be respectively 1.4925 ETH and 0.75 ETH, but instead if sets price as 3 ETH for both sales (2.2425 vs 6 ETH in volume)

**Review by liutaurasjanuskas — CHANGES_REQUESTED** [review]

File: src/adapters/alienswap-ethereum-1/index.ts (hunk 1 above)
Comment (xcaptain) [reply src/adapters/alienswap-ethereum-1/index.ts]: Thanks, fixed, please review again. Currently our contract one token transfer will emit one fill event, I'm now using `considerations[0][3]` I think in this way the price won't recount
image

Comment (liutaurasjanuskas) [reply src/adapters/alienswap-ethereum-1/index.ts]: I get an error: 
`TypeError: event.returnValues.consideration[0][3].dividedBy is not a function`

Comment (xcaptain) [reply src/adapters/alienswap-ethereum-1/index.ts]: Sorry, changed again, the test passed
image

Comment (xcaptain) [reply src/adapters/alienswap-ethereum-1/index.ts]: @liutaurasjanuskas is this pr good for you now?

**Comment by xcaptain** [comment]
@liutaurasjanuskas Thanks for merging this, can you tell me where I can see the trading list?

**Comment by liutaurasjanuskas** [comment]
@xcaptain Hello, it is visible here:

- [NFT marketplaces list](https://dappradar.com/nft/marketplaces/protocol/ethereum)
- [Single dapp page](https://dappradar.com/ethereum/marketplaces/alienswap-2/marketplaces)

**Pull request merged**